Documentation/guides/BindingJavaSource.md

```diff
@@ -0,0 +1,97 @@
+---
+id:
+title: "Binding Java Source"
+dateupdated: 2021-05-21
+---
+
+# Overview
+
+Bindings can be done on not only .jar and .aar files but also raw
+.java code. This will allow developers of bindings or applications
+to write custom API's for underlying bindings and expose them
+easily to the developer.
+
+Applications already had the ability to include `AndroidJavaSource`
+items in them, these would then be compiled into the final `classes.dex`.
+However this custom java source code was not "bound" and if users wanted
+to call these classes they would need to do that manually.
+
+The `AndroidJavaSource` Item Group now supports a `Bind` attribute. This
+will instruct the build system to not only compile the code but also
+produce a C# API for it.
+
+```xml
+<ItemGroup>
+    <AndroidJavaSource Include="MyClass.java" Bind="True" />
+</ItemGroup>
+```
+
+The `Bind` attribute will default to `true` when in binding projects.
+But for applications it will need to be set manually.
+
+# Example
+
+Consider the following Java code.
+
+```java
+package com.xamarin.test;
+
+class MyClass {
+    public bool IsDave (String name)
```

Comment:
[HHOS](https://www.urbandictionary.com/define.php?term=HHOS): should we use one of the [known Java annotations](https://github.com/xamarin/java.interop/blob/a5ed8919fb2ec894cb8144e51ae7c29b4811ee2a/src/Xamarin.Android.Tools.Bytecode/XmlClassDeclarationBuilder.cs#L460-L479) to indicate that `name` shouldn't be null?

```java
    public bool IsDave (@androidx.annotation.NonNull String name)
```